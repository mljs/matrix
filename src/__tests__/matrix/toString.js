import { Matrix } from '../..';

describe('custom toString function', () => {
  it('should work with a simple matrix', () => {
    expect(
      new Matrix([
        [0, 10, 200],
        [-0.3, -0.44, -0.5555],
      ]).toString(),
    ).toMatchSnapshot();
  });

  it('should use clamp options', () => {
    expect(
      new Matrix([
        [10, 200, 3000000, 0, 1],
        [-0.5555, 901569741, 3, 4, 5],
      ]).toString({ maxRows: 1, maxColumns: 3, maxNumSize: 6 }),
    ).toMatchSnapshot();
  });
});
