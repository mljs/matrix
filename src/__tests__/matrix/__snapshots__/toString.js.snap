// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom toString function should use clamp options 1`] = `
"Matrix {
  [
    10     200    3.0e+6 ... 2 more columns
    ... 1 more rows
  ]
  rows: 2
  columns: 5
}"
`;

exports[`custom toString function should work with a simple matrix 1`] = `
"Matrix {
  [
     0        10       200    
    -0.3     -0.44    -0.5555 
  ]
  rows: 2
  columns: 3
}"
`;
